feat: Support for elastic.co in ElasticsearchSriver and improved docs (#2240)
cube-js/cube.js: docs/content/Configuration/Environment-Variables-Reference.md

| `CUBEJS_DB_URL`                                 | Druid, Elasticsearch | The URL for a database                                                                                                                                                                                      | A valid database URL for Druid/Elasticsearch                     |
| `CUBEJS_DB_ELASTIC_OPENDISTRO`                  | Elasticsearch        | If `true`, then use the Open Distro for Elasticsearch                                                                                                                                                       | `true`, `false`                                                  |
| `CUBEJS_DB_ELASTIC_QUERY_FORMAT`                | Elasticsearch        | By default, queries return data in JDBC format, but you can also return data in standard Elasticsearch JDBC, JSON, CSV, YAML or raw formats (only JSON and JDBC are currently supported)                    | `json`, `jdbc`                                                   |

> | `CUBEJS_DB_ELASTIC_APIKEY_ID`                   | Elasticsearch        | Id of the API key to be passed via auth config param. Tested with Elasticsearch instances hosted on elastic.co                                         | Elasticsearch        | A valid Elastic.co API key id                                                                  |
> | `CUBEJS_DB_ELASTIC_APIKEY_KEY`                  | Elasticsearch        | Value of the API key to be passed via auth config param. Tested with Elasticsearch instances hosted on elastic.co                                      | Elasticsearch        | A valid Elastic.co API key value                                                               |

Reviewer: I think it would be better to link out to Elastic's documentation:

```suggestion
| `CUBEJS_DB_ELASTIC_APIKEY_ID`                   | Elasticsearch        | [ID of the API key from elastic.co][link-elastic-api-keys]                                                                                                                                                  | A valid Elastic.co API key ID                                    |
| `CUBEJS_DB_ELASTIC_APIKEY_KEY`                  | Elasticsearch        | [Value of the API key from elastic.co][link-elastic-api-keys]                                                                                                                                               | A valid Elastic.co API key value                                 |
```

Reviewer: Would also need to add this to the end of this file so the linked URL works

```
[link-elastic-api-keys]:
  https://www.elastic.co/guide/en/kibana/master/api-keys.html#create-api-key
```

Author: thanks for finding it, done
